OpenRA.Mods.RA/Traits/GivesTeslaBoost.cs

```diff
@@ -0,0 +1,196 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Mods.RA.Activities;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.RA.Traits
+{
+	public class GivesTeslaBoostInfo : ITraitInfo
+	{
+		[Desc("The upgrades to grant.")]
+		public readonly string[] Upgrades = { "tesla" };
+
+		[Desc("Make this actor grant the upgrades multiple times.")]
+		public readonly int GrantMultiple = 1;
```

Comment:
This guy is here to make Tesla tanks grant `tesla` multiple times because if I remember correctly from RA2, one tank should give more of a boost than a single trooper